### PR TITLE
Tools: Add disruptor

### PR DESCRIPTION
**Pull request by benoit-bst**
A tool to send generic **disruptions** inside **kraken** core for debug purpose
It was very usefull for me to test the reload part.
All the explanations are in the **README.md**
English friends are welcome for the wording errors :cat: 

**Some traces examples**
Launch disruptor to create a disruption on Line 1 - aka line:DUA:100110001
```
PYTHONPATH="/home/bbrisset/dev/navitia/source/chaos-proto" python2.7 /home/bbrisset/dev/navitia/source/scripts/disruption/disruptor.py -b pyamqp://guest:guest@localhost:5672 -e "navitia" -t "shortterm.transilien" -p "Line('line:DUA:100110001')" -i "REDUCED_SERVICE"
2020-03-26 13:56:21,090 - disruptor - INFO - launch disruptor
2020-03-26 13:56:21,090 - disruptor - INFO - create disruption sample for test
2020-03-26 13:56:21,090 - disruptor - INFO - new disruption type: line - uri: line:DUA:100110001 - id: 0bfce4fc-20f9-486f-87dd-6f68fdbba8a2 - start: 2020-03-26 12:56:21.090519 - end: 2020-03-26 13:06:21.090520 - impact type: REDUCED_SERVICE
2020-03-26 13:56:21,090 - disruptor - INFO - send disruption on pyamqp://guest:guest@localhost:5672 - navitia - shortterm.transilien 
2020-03-26 13:56:21,098 - disruptor - INFO - close disruptor
```
Retreive results in jormungandr after the reload

disruption_response

:information_source: I encourage you to follow the set up and test directly the script. Because it is a long road to review it...

### CODE REVIEW

File: source/scripts/disruption/README.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+#############################
+[BROKER]
+host = localhost
+port = 5672
+username = guest
+password = guest
+exchange = navitia
+vhost = /
+
+rt_topics = shortterm.coverage_name
```
Comment (kadhikari) [inline source/scripts/disruption/README.md:36]: You can add "# you should use the same value in command line" above this line.

File: source/scripts/disruption/README.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+protoc chaos.proto gtfs-realtime.proto --python_out=.
+
+# Install RabbitMQ, if it is not done yet
+sudo apt install rabbitmq-server
+```
+
+Update your krkaken.ini with broker and rt_topics
+
+```
+######################### w#####
```
Comment (kadhikari) [inline source/scripts/disruption/README.md]: Some characters to be deleted.

File: source/scripts/disruption/README.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+}
+```
+
+```
+# play file with sleep = 5 sec
+PYTHONPATH="navitia_source_dir/source/chaos-proto" python2.7 disruptor.py -b pyamqp://guest:guest@localhost:5672 -e "navitia" -t "shortterm.transilien" -f path_to_file -s 5
+
+# play file with sleep = 5 sec and loop forever
+PYTHONPATH="navitia_source_dir/source/chaos-proto" python2.7 disruptor.py -b pyamqp://guest:guest@localhost:5672 -e "navitia" -t "shortterm.transilien" -f path_to_file -s 5 -l
+```
```
Comment (kadhikari) [inline source/scripts/disruption/README.md:224]: You my add here:
"# All the object codes in the above command line examples as well as in file 'impacts.josn' will have to be replaced by the object codes existing in your kraken (instance). For a LineSection, StopAreas should be related to the Line.
# You should also use the virtulenv for jormungandr.

**Review by kadhikari — APPROVED** [review]
Great work @benoit-bst 
It's very useful. thanks a lot. :+1:

File: source/scripts/disruption/disruptor.py [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+    def publish(self, producer):
+        producer.publish(self.to_pb())
+
+    def delete(self, producer):
+        self.is_deleted = True
+        producer.publish(self.to_pb())
+
+
+def parse_args(parser, logger):
+    parser = argparse.ArgumentParser()
```
Comment (xlqian) [inline source/scripts/disruption/disruptor.py:213]: next time, if you do argument parsing, you may want to have a look at `docopt`

**Review by xlqian — APPROVED** [review]

File: source/scripts/disruption/README.md (hunk 3 above)
Comment (benoit-bst) [reply source/scripts/disruption/README.md:224]: yes my bad, 
we must have the jormungandr virtualenv to run the script

File: source/scripts/disruption/disruptor.py (hunk 4 above)
Comment (benoit-bst) [reply source/scripts/disruption/disruptor.py:213]: Thanks for the **Tips**. Next time I will be ready :rocket:

File: source/scripts/disruption/disruptor.py [hunk 5]
```diff
@@ -0,0 +1,389 @@
...
+            disruption = Disruption(
+                None,
+                datetime.datetime.utcnow(),
+                datetime.datetime.utcnow() + datetime.timedelta(minutes=args.disruption_duration),
+                logger,
+                args.impact_type,
+            )
+        else:
+            disruption = Disruption(
+                eval(args.pt_object),
```
Comment (mbounabi) [inline source/scripts/disruption/disruptor.py]: Shouldn't there be a check on this arg? Is this present and also, are the info correct? I don't know if it's possible to verify if the pt object is valid or not.

File: source/scripts/disruption/README.md [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+                     [--empty_disruption] [-s SLEEP] [-l]
+
+optional arguments:
+  -h, --help            show this help message and exit
+
+  -b BROKER_CONNECTION, --broker_connection BROKER_CONNECTION
+                        AMQ broker connection string (mandatory). Example: -b
+                        pyamqp://guest:guest@localhost:5672
+
+-e EXCHANGE, --exchange EXCHANGE
```
Comment (mbounabi) [inline source/scripts/disruption/README.md]: Minor: alignment

File: source/scripts/disruption/disruptor.py (hunk 5 above)
Comment (benoit-bst) [reply source/scripts/disruption/disruptor.py]: I use **eval** to convert a `string to pt_pbject`. If the Pt_pbject is not well fomatted, we have just a
```
Traceback (most recent call last):
  File "/home/bbrisset/dev/navitia/source/scripts/disruption/disruptor.py", line 389, in <module>
    main()
  File "/home/bbrisset/dev/navitia/source/scripts/disruption/disruptor.py", line 378, in main
    eval(args.pt_object),
  File "<string>", line 1, in <module>
NameError: name 'Kne' is not defined
```
I can enrich the trace with a **try-catch** and an explicit message.
Thanks for the idea

Comment (benoit-bst) [reply source/scripts/disruption/disruptor.py]: now we have in addition
```
2020-03-30 13:45:31,121 - disruptor - ERROR - pt_object is badly formatted - <type 'exceptions.NameError'>                                                                                                                                                                        
Traceback (most recent call last):                                                                                                                                                                                                                                                
  File "/home/bbrisset/dev/navitia/source/scripts/disruption/disruptor.py", line 398, in <module>                                                                                                                                                                                 
    main()                                                                                                                                                                                                                                                                        
  File "/home/bbrisset/dev/navitia/source/scripts/disruption/disruptor.py", line 384, in main                                                                                                                                                                                     
    eval(args.pt_object),                                                                                                                                                                                                                                                         
  File "<string>", line 1, in <module>                                                                                                                                                                                                                                            
NameError: name 'Kne' is not defined
```

**Review by mbounabi — APPROVED** [review]

**Pull request merged**